### DISCUSSION TITLE
Dropdowns don't appear with `overflow-auto`

### DISCUSSION THREAD

**Discussion by huzaifahj** [formkit/formkit#566] [Ideas]
### Reproduction

https://formkit.link/caaebd1fa7385602b440ac3ad17d4102

### Describe the bug

1. Click icon to open dropdown
2. Dropdown does not open

This is likely because the container for the list items is cut off. Might be fixed by hoisting it to the `<body>`

### Environment

• OS: [e.g. iOS]
• Browser [e.g. chrome, safari]
• Version [e.g. 22]

**Comment by justin-schroeder** [comment]
This is, unfortunately, a limitation of the dropdown. It would resolve itself if it was hoisted to the body and we considered doing so when we were designing the input’s architecture, but doing so leads to a number of other issues that we decided would lead to a poorer DX than this shortcoming.

We don’t have any immediate plans to fix this, but it’s worth keeping alive as a feature request under our discussions.

**Reply by huzaifahj** [reply]
Thanks @justin-schroeder . Wondering what the other issues are and if it could be added via a setting instead? I agree that the default should have the best DX. 

The non-native inputs like these are great but as it doesn't work with our UI (working with overflow is quite common) we can't use it

**Comment by smakinson** [comment]
@justin-schroeder I'm looking into this issue in our project as well. We are using Quasar, which has its own portal setup for its components that have a popup of any sort, but I'm trying to stick with FormKit for inputs in this project for consistency & easier schema use. I understand the dilemma mentioned her & I found more detail in Discord. I have a question that MIGHT partially solve the issue around dropdown's. Have you investigated an input that uses the native input like the `select` but has features found in `dropdown`, etc? Perhaps it would also be a `pro` component.  I think the biggest issue to work out would be how to allow user input for searching and also clicking to open the native select. Is this a path that could work for that component? (which is likely used way more often than the others like color & date)

Another benefit to a component like this would be it would feel a bit more consistent if we wanted to mix the use of `select` with the new component, that way the list of options would not seem so different from each other.

Even if this new component did not pick up the search capability, it would be nice for it to have the features for using a function to load options, etc. Perhaps that would be a good first step if this is feasible?

**Reply by justin-schroeder** [reply]
Unfortunately there is no way to programtically (cross browser) open a `<select>` input — so while it would be possible to use the select the user would need to manually "open" the select after searching. We decided against this due to the poor UX.

However there is some hope on the horizon! The new popover API should categorically solve these issues. For example: https://codepen.io/justin-schroeder/pen/VwqVROw

However browser support leaves something to be desired:

![image](https://github.com/formkit/formkit/assets/1950463/f07f97f6-011a-40e3-aec1-fdc1468188dc)

Still — exciting to see this on the horizon.

**Reply by smakinson** [reply]
Ya that will be nice. Any thoughts on a pro `select` with the features that work out and are unrelated to the user input such as: https://formkit.com/inputs/dropdown#dynamic-options ?

**Comment by andrew-boyd** [comment]
As of `0.120.0` which shipped today you can use the `popover` prop on the `autocomplete`, `colorpicker`, `datepicker`, `dropdown` and `taglist` inputs to render their contents using the browser Popover API. Firefox does not _yet_ support popover but will in upcoming version `125` — current version is `122`

https://caniuse.com/?search=popover

Demo: https://formkit.link/2332f8765d4fb6b7424b8ce6252b8f49
